commit 6ddcfdcf919463469496132731671ebb5f9908e9
Author: Greg Kroah-Hartman <gregkh@suse.de>
Date:   Thu Oct 14 12:26:08 2010 -0700

    Staging: brcm80211: fix up compiler warnings I created
    
    I messed up on some of the uintptr conversions, this fixes them up
    properly.
    
    Cc: Brett Rudley <brudley@broadcom.com>
    Cc: Henry Ptasinski <henryp@broadcom.com>
    Cc: Nohee Ko <noheek@broadcom.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

--- a/drivers/staging/brcm80211/util/aiutils.c
+++ b/drivers/staging/brcm80211/util/aiutils.c
@@ -134,7 +134,7 @@ void ai_scan(si_t *sih, void *regs, uint
 	case SPI_BUS:
 	case SDIO_BUS:
 #endif				/* BCMSDIO */
-		eromptr = (u32 *)erombase;
+		eromptr = (u32 *)(unsigned long)erombase;
 		break;
 
 	default:
@@ -358,8 +358,8 @@ void *ai_setcoreidx(si_t *sih, uint core
 	case SPI_BUS:
 	case SDIO_BUS:
 #endif				/* BCMSDIO */
-		sii->curmap = regs = (void *)addr;
-		sii->curwrap = (void *)wrap;
+		sii->curmap = regs = (void *)(unsigned long)addr;
+		sii->curwrap = (void *)(unsigned long)wrap;
 		break;
 
 	default:
